shared-libs/cht-datasource/src/libs/contact.ts

```diff
@@ -0,0 +1,15 @@
+import { Doc } from './doc';
+import { DataObject } from './core';
+
+interface DehydratedParent extends DataObject {
+  _id: string;
+  parent?: DehydratedParent;
+}
+
+export interface Contact extends Doc {
+  contact_type?: string;
+  name?: string;
+  parent?: DehydratedParent;
+  reported_date?: Date;
+  type: string;
+}
```

Comment:
So, my thought with these interfaces is that conceptually they should not depend on the db (or on data needed for querying the db). Instead they just represent the contract with any consumers of cht-datasource of what we _guarantee_.  Unfortunately, for `V1` of the API, we cannot really guarantee anything since the existing code we are trying to replace does not do much schema validation when looking up data.  That is why I ended up making most of these fields optional here since the existing code does not do any kind of checking that the contact actually has a `name` value.  In `V2`, we should definitely come up with a list of _required_ fields that we would check for when pulling data from the DB. Then our V2 interfaces can guarantee those fields to be valid.

Does that make sense, or have I missed your point entirely?  :sweat_smile:

Reply:
Yes it totally makes sense, thanks for sharing your thought process here :raised_hands: